projects/solace-message-client/src/lib/solace-message-client.ts

```diff
@@ -201,110 +355,88 @@ export interface PublishOptions {
    * When this property is set, when the message expires in the network,
    * the message is saved on a appliance dead message queue. Otherwise the expired message is
    * discarded. See {@link PublishOptions.setTimeToLive}.
-   *
-   * Note: Property is ignored if providing a {@link solace.Message} payload object.
    */
   dmqEligible?: boolean;
 
   /**
    * Sets the delivery mode of the message.
    *
-   * Note: Property is ignored if providing a {@link solace.Message} payload object.
+   * Solace supports two delivery modes, also known as qualities of service (QoS):
+   * - Direct Messaging (default if not specified)
+   * - Guaranteed or Persistent Messaging
+   *
+   * For a detailed comparison, refer to https://solace.com/blog/delivery-modes-direct-messaging-vs-persistent-messaging.
+   *
+   * ### Direct Messaging (default)
+   * Direct messaging is over TCP and provides a reliable, but not guaranteed, delivery of messages from the Solace event broker to consuming clients,
+   * and is the default message delivery system. The broker delivers a message one time at most. It is most appropriate for messaging applications
+   * that require very high-rate or very low-latency message transmission, i.e., to efficiently publish messages to a large number of clients with matching
```

Comment:
As hint for you, other apis for example java api supports:
https://solace.com/blog/understanding-guaranteed-message-publish-window-sizes-and-acknowledgement/

The you can achieve the same speed with guaranteed messaging. In JS the windows size is always 1

Reply:
Thank you for the hint, added it to the JS doc.

Reply:
_If you experience poor performance when sending guaranteed messages, consider adjusting the size of the publishing window via {​​​​​​@link Session#publisherProperties#windowSize}​​​​​​, as described here: https://solace.com/blog/understanding-guaranteed-message-publish-window-sizes-and-acknowledgement_